source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1441,6 +1441,70 @@ void RtFileIo::writeDefinitions(const FilePath& documentPath, const RtTokenVec&
     writeDefinitions(ofs, names, options);
 }
 
+RtPrim RtFileIo::readPrim(std::istream& stream, const RtReadOptions* options)
+{
+    try
+    {
+        DocumentPtr document = createDocument();
+        XmlReadOptions xmlReadOptions;
+        if (options)
+        {
+            xmlReadOptions.applyFutureUpdates = options->applyFutureUpdates;
+        }
+        readFromXmlStream(document, stream, &xmlReadOptions);
+
+        PvtStage* stage = PvtStage::ptr(_stage);
+
+        RtReadOptions::ElementFilter filter = options ? options->elementFilter : nullptr;
+
+        // Keep track of renamed nodes:
+        ElementPtr elem = document->getChildren()[0];
```

Comment:
You should check that at least one child was loaded, before dereferencing it here.